app/models/concerns/media_facebook_item.rb

```diff
@@ -200,7 +205,7 @@ def data_from_facebook_item
       self.data['text'].strip!
       self.data['media_count'] = 1 unless self.url.match(/photo\.php/).nil?
       self.data.merge!({
-        username: self.data['user_name'],
+        username: self.get_facebook_slug_from_html || self.data['user_name'],
```

Comment:
Perhaps this should return an empty string instead of falling back to the real name when username is unknown, given we already have a dedicated field for the user's real name?